src/engine/MaxConstraint.cpp

```diff
@@ -311,15 +346,31 @@ void MaxConstraint::getEntailedTightenings( List<Tightening> &tightenings ) cons
             {
                 if ( !existsUpperBound( element ) ||
                      FloatUtils::gt( getUpperBound( element ), fUB ) )
-                    tightenings.append
-                        ( Tightening( element, fUB, Tightening::UB ) );
+                {
+                    if ( proofs )
+                    {
+                        ASSERT( _elementToTighteningRow[element] != NULL );
+                        _boundManager->tightenUpperBound( element, fUB, *_elementToTighteningRow[element] );
```

Comment:
This is a significant divergence in behavior between the case when proofs are on and when they are off. This function doesn't register bound tightenings, it only computes them.

Reply:
Agree. However, for proof production it is crucial that explanations and bounds are updated as soon as they are discovered, since otherwise we might have timing problems (as we previously had in the ReLU case).